docs/guides/jupyter.rst

```diff
@@ -0,0 +1,363 @@
+refer to the nbsphinx_ or MyST-NB_ documentation.
+
+Rendering interactive widgets
+-----------------------------
+
+Widgets are eventful python objects that have a representation in the browser
+and that you can use to build interactive GUIs for your notebooks.
+Basic widgets using `ipywidgets`_ include controls like sliders, textboxes, and buttons,
+and more complex widgets include interactive maps, like the ones provided by `ipyleaflet`_.
+
+.. _ipywidgets: https://ipywidgets.readthedocs.io/
+.. _ipyleaflet: https://ipyleaflet.readthedocs.io/
+
+You can embed these interactive widgets on HTML Sphinx documentation.
+For this to work, it's necessary to *save the widget state*
+before generating the HTML documentation,
+otherwise the widget will appear as empty.
+Each editor has a different way of doing it:
+
+- The classical Jupyter Notebook interface
+  provides a "Save Notebook Widget State" action in the "Widgets" menu,
+  :ref:`as explained in the ipywidgets
+  documentation <ipywidgets:/embedding.md#embedding-widgets-in-html-web-pages>`.
+  You need to click it before exporting your notebook to HTML.
+- JupyterLab provides a "Save Widget State Automatically" option in the "Settings" menu.
+  You need to leave it checked so that widget state is automatically saved.
+- In Visual Studio Code `it's not possible to save the widget
+  state <https://github.com/microsoft/vscode-jupyter/issues/4404>`_
+  at the time of writing.
+
+.. figure:: /_static/images/guides/jupyterlab-save-widget-state.png
+   :width: 30%
+   :align: center
+   :alt: JupyterLab option to save the interactive widget state automatically
+
+   JupyterLab option to save the interactive widget state automatically
+
+For example, if you create a notebook with a simple
+:ref:`ipywidgets:/examples/widget list.ipynb#intslider`
+widget from ipywidgets and save the widget state,
+the slider will render correctly in Sphinx.
+
+.. figure:: /_static/images/guides/widget-html.gif
+   :width: 80%
+   :align: center
+   :alt: Interactive widget rendered in HTML by Sphinx
+
+   Interactive widget rendered in HTML by Sphinx
+
+To see more elaborate examples:
+
+- `ipyleaflet`_ provides several widgets for interactive maps,
+  and renders live versions of them `in their
+  documentation <https://ipyleaflet.readthedocs.io/en/latest/api_reference/velocity.html>`_.
+- `PyVista <https://docs.pyvista.org/>`_ is used for scientific 3D visualization
+  with several interactive backends and `examples in their
+  documentation <https://docs.pyvista.org/index.html#maps-and-geoscience>`_ as well.
+
+.. warning::
+
+   Although widgets themselves can be embedded in HTML as shown above,
+   :doc:`events <ipywidgets:examples/Widget Events>`
+   require a backend (kernel) to execute.
+   Therefore, ``@interact``, ``.observe``, and related functionalities relying on them
+   will not work as expected.
+
+.. _notebook-galleries:
+
+Creating galleries of examples using notebooks
```

Comment:
to be honest this one feels a little bit out-of-place given the tutorial's topic. I feel like "generate a gallery from multiple pages" is a generic use-case that is not notebook-specific (though it is a cool feature of nbsphinx). I almost feel like it deserves its own section (that covers `sphinx-gallery` primarily, and links to `nbsphinx` to talk about notebook-specific functionality) rather than to talk about galleries in the context of notebooks.

Reply:
I'm not sure. While "generating galleries from notebooks" is something that several projects are doing or seeking to do, "generating galleries from multiple pages" sounds like too abstract of a use case to have a guide dedicated to it.

For me as a user, creating galleries is 40 % of the reason why notebooks are cool, since thumbnails can be automatically generated and notebooks are an excellent choice for creating how-to guides. About sphinx-gallery, I tried using it in a personal project that was using notebooks for examples already https://github.com/poliastro/poliastro/pull/828 but the workflow ended up being more complex, so we abandoned this approach.

Reply:
I'll leave this one to you - I think it comes down to whether "creating a gallery of notebooks" is a "core part of the notebook experience". In my opinion, it isn't, but I don't feel strongly about it, so if others believe strongly that it _is_, then I'll defer

Reply:
I had the same feedback -- I think we probably want a different guide on galleries, and to link from that from this page. https://github.com/readthedocs/readthedocs.org/pull/8283#discussion_r658350299

Reply:
Fair enough, perhaps I was being too stubborn - I agree to make another guide about example galleries. However, given that I would need to learn sphinx-gallery and investigate if there are other alternatives, do you folks think we can do it in another pull request? That way we can get this closer to being published.